### ISSUE TITLE
DateIndexFormatterTest fails on my timezone

### ISSUE THREAD

**Issue by takezoe** [elastic/elasticsearch-hadoop#949]
In my local environment (default timezone is JST), [DateIndexFormatterTest](https://github.com/elastic/elasticsearch-hadoop/blob/b7ea331ebf4727b48e789e4ddd75e24c221602a2/mr/src/test/java/org/elasticsearch/hadoop/serialization/field/DateIndexFormatterTest.java) fails as following:

```
org.elasticsearch.hadoop.serialization.field.DateIndexFormatterTest > testTimeYMDFormat FAILED
    java.lang.AssertionError at DateIndexFormatterTest.java:34
```

In details, expected: is "2014.10.06" but: was "2014.10.07".

I would like to fix this test, but I don't know what is an expected behavior of [DateIndexFormatter](https://github.com/elastic/elasticsearch-hadoop/blob/b7ea331ebf4727b48e789e4ddd75e24c221602a2/mr/src/main/java/org/elasticsearch/hadoop/serialization/field/DateIndexFormatter.java). Is it always expected to return UTC date?

In addition, this test succeeds with JDK based date parsing. Why is date parsing switchable in [DateUtils](https://github.com/elastic/elasticsearch-hadoop/blob/b7ea331ebf4727b48e789e4ddd75e24c221602a2/mr/src/main/java/org/elasticsearch/hadoop/util/DateUtils.java)?

**Comment by jbaiera** [comment]
@takezoe I'll dig around a bit to see if I can provide an answer to what's expected. Timezones are a tricky subject to get right. As for why there exists a runtime bridge in DateUtils: My guess is because JodaTime has proven to be generally better at parsing date information than the facilities that are available in the JDK, often times supporting a wider amount of the available standards.

**Comment by takezoe** [comment]
> Timezones are a tricky subject to get right.

Exactly. The point in this case is that it's not clear what timezone `DateIndexFormatter` should use to format a given date. Is there any problem if it always use UTC?

> JodaTime has proven to be generally better at parsing date information than the facilities that are available in the JDK

Thanks for explanation, understood.

**Comment by cwensel** [comment]
It isn't intuitive to me to be indexing time series data logged in UTC, but have index names generated in the local timezone. In practice (if you happen to be in the pacific coast) there will be an index named `logs-2017.12.31` with data having timestamps in 2018. 

I'm ok if this translates to be an option to turn this on so that current users don't see a gap in data. the option could be simply the timezone to use when formatting the index string.
